code/data_structures/src/list/singly_linked_list/operations/insertion/insertion_at_head.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+  C++ program to add elements at the head of a Singly Linked List
+  Part of Cosmos by OpenGenus Foundation
+ */
+
+#include<iostream>
+#include<cstdlib>
+
+using namespace std;
+
+
+struct node{
+             int data;
+             node *next;
+
+            };
+
+void printList(node *head)
+{
+    cout<<"This is your current List\n";
+    while(head!=NULL)
+    {
+        cout<<head->data<<" => ";
+        head=head->next;
+    }
+    cout<<"NULL\n";
+}
+
+
+// Function to add A node at the Head of the List
+void insertAtHead(node **head,int data)
+{
```

Comment:
Include a case or just a check when `head` is NULL to make sure the user uses this function correctly